kahuna/public/js/metadata-templates/metadata-templates.js

```diff
@@ -95,27 +142,40 @@ metadataTemplates.controller('MetadataTemplatesCtrl', [
 
   ctrl.applyTemplate = () => {
     ctrl.saving = true;
+    let promise = Promise.resolve();
 
-    editsService
-      .update(ctrl.image.data.userMetadata.data.metadata, ctrl.metadata, ctrl.image)
-      .then(resource => ctrl.resource = resource)
-      .then(() => {
-        if (ctrl.metadataTemplate.collectionFullPath) {
-          collections.addCollectionToImage(ctrl.image, ctrl.metadataTemplate.collectionFullPath);
-        }
-      })
-      .then(() => {
-        if (ctrl.metadataTemplate.usageRights) {
-          editsService
-            .update(ctrl.image.data.userMetadata.data.usageRights, ctrl.metadataTemplate.usageRights, ctrl.image)
-            .then(() => editsService.updateMetadataFromUsageRights(ctrl.image, false));
+    promise.then(() => {
+      if (ctrl.metadataTemplate.metadataFields.length > 0) {
+        editsService
+          .update(ctrl.image.data.userMetadata.data.metadata, applyTemplateToMetadata(ctrl.metadataTemplate.metadataFields), ctrl.image);
         }
       })
-      .finally(() => {
-        ctrl.metadataTemplate = null;
-        ctrl.saving = false;
-        ctrl.onMetadataTemplateApplied();
-      });
+    .then(() => {
+      if (ctrl.metadataTemplate.collectionFullPath) {
+        collections.addCollectionToImage(ctrl.image, ctrl.metadataTemplate.collectionFullPath);
+      }
+    })
+    .then(() => {
+      if (ctrl.metadataTemplate.usageRights) {
+        editsService
+          .update(ctrl.image.data.userMetadata.data.usageRights, applyTemplateToUsageRights(ctrl.metadataTemplate.usageRights), ctrl.image)
+          .then(() => editsService.updateMetadataFromUsageRights(ctrl.image, false));
+      }
+    })
+    .then(() => {
+      if (ctrl.metadataTemplate.lease) {
+        let lease = toLease(ctrl.metadataTemplate.lease);
+        leaseService.batchAdd(lease, [ctrl.image]);
+      }
+    })
```

Comment:
I think these should happen separately after the previous `then` finishes? if so you'll need to make sure to return the promise from each `then` (eg. `return collections.addCollectionToImage(...)`) otherwise js will immediately move onto the next without waiting for completion

Reply:
Good catch. Thank you 👍🏽